CONTRIBUTING.md

```diff
@@ -32,63 +32,246 @@ Follow [@zitadel](https://twitter.com/zitadel) on twitter
 
-Ensure the provided code meets the [official style guide](https://developers.google.com/protocol-buffers/docs/style).
+  1. Install `VcXsrv Windows X Server`
+  2. Set the target of your shortcut to `"C:\Program Files\VcXsrv\xlaunch.exe" -ac`
+  3. In WSL2 run `export DISPLAY=$(cat /etc/resolv.conf | grep nameserver | awk '{print $2}'):0` to set your DISPLAY variable
+  4. When starting XLaunch, make sure to disable access control
+</details>
 
-The following docker command builds the grpc stub into the correct folders:
+Run the database and the latest backend locally.
 
 ```bash
-docker build -f build/grpc/Dockerfile -t zitadel-base:local . \
-    && docker build -f build/zitadel/Dockerfile . -t zitadel-go-base --target go-copy -o .
+# Change to the console directory
+cd ./console
+
+# You just need the db and the zitadel services to develop the console against.
+docker compose --file ../e2e/docker-compose.yaml up --detach db zitadel
 ```
 
-## Contribute Docs
+When the backend is ready, you have the latest zitadel exposed at http://localhost:8080.
+You can now run a local development server with live code reloading at http://localhost:4200.
+To allow console access via http://localhost:4200, you have to configure the ZITADEL backend.
 
-<!-- TODO: ask maxpe for infos -->
+1. Navigate to http://localhost:8080/ui/console/projects.
+2. When prompted, login with *zitadel-admin@<span because="breaks the mailto"></span>zitadel.localhost* and *Password1!*
+3. Select the *ZITADEL* project.
+3. Select the *Console* application.
+4. Select *Redirect Settings*
+5. Add *http://<span because="breaks the link"></span>localhost:4200/auth/callback* to the *Redirect URIs*
+6. Add *http://<span because="breaks the link"></span>localhost:4200/signedout* to the *Post Logout URIs*
+7. Select the *Save* button
 
-We add the label "good first issue" for problems we think are a good starting point to contribute to ZITADEL.
+You can run the local console development server now.
 
-- [Issues for first time contributors](https://github.com/zitadel/zitadel/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22)
-- [All issues](https://github.com/zitadel/zitadel/issues)
+```bash
+# Console loads its target environment from the file console/src/assets/environment.json.
+# Load it from the backend.
+curl http://localhost:8080/ui/console/assets/environment.json > ./src/assets/environment.json
+
+# Generate source files from Protos
+npm run generate
+
+# Install npm dependencies
+npm install
+
+# Start the server
+npm start
+```
+
+Navigate to http://localhost:4200/.
+Make some changes to the source code and see how the browser is automatically updated.
+After making changes to the code, you should run the end-to-end-tests.
+Open another shell.
+
+```bash
+# Change to the e2e directory
+cd ./e2e
```

Comment:
```suggestion
cd .. && cd e2e/
```

Reply:
Above it says *Open another shell*. Should we assume you are still in the same directory with the new shell?